.github/workflows/branch_preparation.yml

```diff
@@ -0,0 +1,116 @@
+name: Branch Preparation CI
+
+on:
+  push:
+    branches:
+      - 'release/**'
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+**'
+
+jobs:
+  prepare_develop_branch_after_release:
+    runs-on: ubuntu-latest
+    if: ${{ github.ref_type == 'tag' }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: |
+          git checkout develop
+
+      - uses: cachix/install-nix-action@v22
+      - name: Pre-populate nix-shell
+        run: |
+          export NIX_PATH=nixpkgs=$(jq '.nixpkgs.url' nix/sources.json -r)
+          echo "NIX_PATH=$NIX_PATH" >> $GITHUB_ENV
+          nix-shell --pure --run "echo" ./shell.nix
+      - name: Check if the chart is publishable
+        run: |
+            tag=${{ github.ref_name }}
+            echo "BASE=$(nix-shell --pure --run "./scripts/update-chart-version.sh --tag $tag --type develop" ./shell.nix)" >> $GITHUB_ENV
```

Comment:
What happens if this is a patch release? In this case we don't want to modify develop, is this taken care of?
Maybe add some simple test cases for the update script to make it clear what is meant to happen for different releases

Reply:
Yes, if it's a patch release then the minor version bump for the develop will still be same, so will be no op for develop.

Reply:
I have added the tests.